client/components/app.jsx

```diff
@@ -79,17 +87,60 @@ class App extends React.Component {
       .catch(error => console.error('Error:', error));
   }
 
+  getOneGrade(gradeId) {
+    fetch(`/api/grades/${gradeId}`)
+      .then(resp => resp.json())
+      .then(data => {
+        return this.setState(grade => ({ grade: data }));
+      })
+      .catch(error => console.error('Error:', error));
+    setTimeout(() => { this.toggleTrue(); }, 500);
```

Comment:
There are not very many big no-nos in coding, but this is one of them. You don't want to write code that waits an arbitrary about of time for something asynchronous to finish. Are you trying to do this?

```js
.then(data => {
  return this.setState(() => ({ grade: data, currentlyEditing: true }));
})

```

Or something different?